tests/test_base.py

```diff
@@ -423,6 +424,51 @@ def test_environ_overrides_default(self):
         with Environ(TESTNS_SECTION1_FOO='blah'):
             self.assertEqual('blah', getter.getstr('section1.foo'))
 
+    def test_config_to_INI(self):
+        getter = getconf.ConfigGetter('TESTNS')
+        getter.getstr('foo', default='bar', doc='Foo\nbar')
+        getter.getlist('section2.foo', default=['first', 'second'])
+        getter.getlist('section2.bar', default=[], doc='Emptylist')
+        getter.getbool('section3.bar', default=True, doc='Important bool value')
+
+        self.assertEqual(getter.get_ini_template(), """\
+[DEFAULT]
+; TESTNS_FOO - type=str - Foo bar
+;foo = bar
+
+[section2]
+; TESTNS_SECTION2_BAR - type=list - Emptylist
+;bar =
+; TESTNS_SECTION2_FOO - type=list
+;foo = first, second
+
+[section3]
+; TESTNS_SECTION3_BAR - type=bool - Important bool value
+;bar = on""")
+
+    def test_reload_uncommented_template(self):
+        getter = getconf.ConfigGetter('TESTNS')
+        getter.getstr('foo', default='foo bar', doc='Foo\nbar')
+        getter.getlist('section2.some_list', default=['first', 'second'])
+        getter.getlist('section2.empty_list', default=[], doc='Emptylist')
+        getter.getbool('section3.true_bool', default=True, doc='Important bool value')
+        content = getter.get_ini_template()
+        with tempfile.NamedTemporaryFile('wt', delete=False) as f:
+            for line in content.splitlines():
+                if line.startswith(';') and not line.startswith('; '):
+                    # Only uncomment entries, not comment
+                    line = line[1:]
+                f.write(line + '\n')
+
+        other_getter = getconf.ConfigGetter('TESTNS', config_files=(f.name,))
```

Comment:
Nitpicking but an hypothetic error in `ConfigGetter.__init__` would leave a temporary file.

Why not the whole block inside the context and get rid of this `try/except`and `unlink` already done by `NamedTemporaryFile(delete=True)`?